llvm/lib/Target/AArch64/Disassembler/AArch64Disassembler.cpp

```diff
@@ -741,6 +744,18 @@ static DecodeStatus DecodeMatrixTile(MCInst &Inst, unsigned RegNo,
   return Success;
 }
 
+static DecodeStatus DecodePPRorPNRRegisterClass(MCInst &Inst, unsigned RegNo,
```

Comment:
Cleanup suggestion for a separate patch:

All these decoder classes looks rather identical, it would be nice to clean this up with something like this:

```
template <unsigned RegClassID, unsigned FirstReg, unsigned NumRegsInClass>
static DecodeStatus DecodeSimpleReg(MCInst &Inst, unsigned RegNo,
                                                uint64_t Addr,
                                                const MCDisassembler *Decoder) {
  if (RegNo > (NumRegsInClass-1))
    return Fail;

  unsigned Register =
      AArch64MCRegisterClasses[RegClassID].getRegister(RegNo);
  Inst.addOperand(MCOperand::createReg(Register + FirstReg));
  return Success;
}
```

And then remove the other functions and just specify e.g. `DecodeSimpleReg<AArch64::PPRorPNRRegClassID, 0, 16>` in the .td file.

Reply:
That sounds good to me. We can do it in a separate patch.